src/elements/content-sidebar/activity-feed/version/CollapsedVersion.js

```diff
@@ -13,21 +13,40 @@ import messages from '../../../common/messages';
 import selectors from '../../../common/selectors/version';
 import { ACTIVITY_TARGETS } from '../../../common/interactionTargets';
 import type { User, FileVersions } from '../../../../common/types/core';
+import { ACTION_TYPE_CREATED, ACTION_TYPE_RESTORED, ACTION_TYPE_TRASHED } from '../../../../constants';
 import './Version.scss';
 
+const ACTION_MESSAGE_UPLOAD = 'uploaded';
+const ACTION_MESSAGE_RESTORE = 'restored';
+const ACTION_MESSAGE_TRASH = 'deleted';
+
 function getMessageForAction(
     action: string,
-    collaborators: { [collaborator_id: string]: User },
+    collaborators: { [collaborator_id: string]: User } = {},
     version_start: number,
     version_end: number,
+    action_by?: User[],
 ): React.Node {
-    // We only support collapsing for multiple upload versions
-    if (action !== 'upload') {
+    if (
+        action !== 'upload' &&
+        action !== ACTION_TYPE_RESTORED &&
+        action !== ACTION_TYPE_TRASHED &&
+        action !== ACTION_TYPE_CREATED
```

Comment:
I feel like a switch here with a default to null would read better as well or invert this gate and default to returning null.